src/sqlancer/mysql/oracle/MySQLDQEOracle.java

```diff
@@ -0,0 +1,498 @@
+        String tableName = tables.getTables().stream().map(AbstractTable::getName).collect(Collectors.joining(","));
+
+        // DQE does not support aggregate functions, windows functions
+        // This method does not generate them, may need some configurations if they can be generated
+        MySQLExpressionGenerator expressionGenerator = new MySQLExpressionGenerator(state)
+                .setColumns(tables.getColumns());
+        MySQLExpression whereClause = expressionGenerator.generateExpression();
+
+        // MySQLVisitor is not deterministic, we should keep it only once.
+        // Especially, in MySQLUnaryPostfixOperation and MySQLUnaryPrefixOperation
+        String whereClauseStr = MySQLVisitor.asString(whereClause);
+
+        String selectStmt = generateSelectStatement(tables, tableName, whereClauseStr);
+
+        String updateStmt = generateUpdateStatement(tables, tableName, whereClauseStr);
+
+        String deleteStmt = generateDeleteStatement(tables, tableName, whereClauseStr);
+
+        for (MySQLTable table : tables.getTables()) {
+            addAuxiliaryColumns(table);
+        }
+
+        state.getState().getLocalState().log(selectStmt);
+        SQLQueryResult selectExecutionResult = executeSelect(selectStmt, tables);
+        state.getState().getLocalState().log(selectExecutionResult.getAccessedRows().values().toString());
+        state.getState().getLocalState().log(selectExecutionResult.getQueryErrors().toString());
+
+        state.getState().getLocalState().log(updateStmt);
+        SQLQueryResult updateExecutionResult = executeUpdate(updateStmt, tables);
+        state.getState().getLocalState().log(updateExecutionResult.getAccessedRows().values().toString());
+        state.getState().getLocalState().log(updateExecutionResult.getQueryErrors().toString());
+
+        state.getState().getLocalState().log(deleteStmt);
+        SQLQueryResult deleteExecutionResult = executeDelete(deleteStmt, tables);
+        state.getState().getLocalState().log(deleteExecutionResult.getAccessedRows().values().toString());
+        state.getState().getLocalState().log(deleteExecutionResult.getQueryErrors().toString());
+
+        String compareSelectAndUpdate = compareSelectAndUpdate(selectExecutionResult, updateExecutionResult);
+        String compareSelectAndDelete = compareSelectAndDelete(selectExecutionResult, deleteExecutionResult);
+        String compareUpdateAndDelete = compareUpdateAndDelete(updateExecutionResult, deleteExecutionResult);
+
+        String errorMessage = compareSelectAndUpdate == null ? "" : compareSelectAndUpdate + "\n";
+        errorMessage += compareSelectAndDelete == null ? "" : compareSelectAndDelete + "\n";
+        errorMessage += compareUpdateAndDelete == null ? "" : compareUpdateAndDelete + "\n";
+
+        if (!errorMessage.isEmpty()) {
+            throw new AssertionError(errorMessage);
+        } else {
+            state.getState().getLocalState().log("PASS");
```

Comment:
I think it would be better to remove this to not clutter the log files with non-essential information.